cmd/pr-reminder/main.go

```diff
@@ -358,9 +358,25 @@ func main() {
 					logger := logrus.WithFields(logrus.Fields{
 						"kerberosId": user.KerberosId,
 					})
-					if err = sendMessage(logger, user.SlackId, user.PrRequests, slackClient); err != nil {
-						logger.WithError(err).Error("failed to message user")
-						errs = append(errs, err)
+
+					const maxPRsPerMessage = 40
+					if len(user.PrRequests) > maxPRsPerMessage {
+						for i := 0; i < len(user.PrRequests); i += maxPRsPerMessage {
+							end := i + maxPRsPerMessage
+							if end > len(user.PrRequests) {
+								end = len(user.PrRequests)
+							}
+							prChunk := user.PrRequests[i:end]
+							if err = sendMessage(logger, user.SlackId, prChunk, slackClient); err != nil {
+								logger.WithError(err).Error("failed to message user")
+								errs = append(errs, err)
+							}
+						}
+					} else {
+						if err = sendMessage(logger, user.SlackId, user.PrRequests, slackClient); err != nil {
```

Comment:
Since the sendMessage will happen either way, it would be better to move the logic in that function. In addition, the chunk PR operation needs its function with unittests to ensure the logic. In the end, in the main, the logic will remain the same, and the `sendMessage`  will ensure sending the message for each chunk with the logic tested.

Reply:
Updated.